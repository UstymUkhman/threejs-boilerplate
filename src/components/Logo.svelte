<div>
  <img src={Logo} alt="Three.js" />
  <h1>Three.js Boilerplate</h1>
</div>

<script lang="ts">
  import Logo from '/img/logo.png';
</script>

<style lang="scss">
  @use "@scss/variables.scss" as var;

  div {
    box-shadow: 0px 0px 5px var.$dark;
    backdrop-filter: blur(5px);

    justify-content: center;
    pointer-events: none;

    justify-self: center;
    align-items: center;

    border-radius: 10px;
    position: absolute;
    padding: 25px 50px;

    display: flex;
    margin: auto;

    height: 50px;
    width: 350px;

    bottom: 0;
    right: 0;
    left: 0;
    top: 0;

    img {
      margin-right: 15px;
      max-width: 50px;
      height: auto;
    }

    h1 {
      font-family: "Roboto Mono", monospace;
      color: var.$dark;
      font-size: 20px;
    }
  }
</style>
