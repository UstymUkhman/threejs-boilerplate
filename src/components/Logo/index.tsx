import LogoImage from '@/assets/images/logo.png';
import Style from './Logo.module.scss';

export const Logo = () =>
(
  <div class={Style.logo}>
    <img src={LogoImage} alt="Three.js" />
    <h1>Three.js Boilerplate</h1>
  </div>
);
