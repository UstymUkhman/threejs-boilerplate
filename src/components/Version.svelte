<aside>v{import.meta.env.BUILD}</aside>

<style lang="scss">
  @use "@scss/variables.scss" as var;

  aside {
    background-color: var.$dark;
    pointer-events: none;

    padding: 3px 10px;
    color: var.$light;
    font-size: 12px;

    position: fixed;
    display: block;

    bottom: 0;
    right: 0;
  }
</style>
