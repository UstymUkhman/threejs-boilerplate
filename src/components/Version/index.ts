export { default as Version } from './Version';
