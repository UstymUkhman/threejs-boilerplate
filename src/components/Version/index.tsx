import Style from './Version.module.scss';

export const Version = () => <aside class={Style.version}>v{VERSION}</aside>;
