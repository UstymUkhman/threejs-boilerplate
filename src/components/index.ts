import { Logo } from '@/components/Logo';
import { Version } from '@/components/Version';

export { Logo, Version };
