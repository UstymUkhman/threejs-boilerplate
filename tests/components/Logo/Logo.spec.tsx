import { Logo } from '@/components';
import { render } from 'solid-testing-library';
import { describe, expect, test } from 'vitest';

describe('<Logo />', () => {
  test('Render', () => {
    const { container, unmount } = render(() => <Logo />);
    expect(container.innerHTML).toMatchSnapshot();
    unmount();
  });
});
