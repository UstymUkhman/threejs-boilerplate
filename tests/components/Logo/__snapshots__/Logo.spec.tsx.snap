// Vitest Snapshot v1

exports[`<Logo /> > Render 1`] = `"<div class=\\"_logo_1j0s2_1\\"><img alt=\\"Three.js\\" src=\\"/img/logo.png\\"><h1>Three.js Boilerplate</h1></div>"`;
